tools/clang_tidy/lib/src/options.dart

```diff
@@ -10,6 +10,16 @@ import 'package:path/path.dart' as path;
 // Path to root of the flutter/engine repository containing this script.
 final String _engineRoot = path.dirname(path.dirname(path.dirname(path.dirname(path.fromUri(io.Platform.script)))));
 
+
+/// Adds warnings as errors for only specific runs.  This is helpful if migrating one platform at a time.
+String? _platformSpecificWarningsAsErrors(String targetVariant) {
+  if (targetVariant == 'host_debug' && io.Platform.isMacOS) {
+    return 'performance-move-const-arg,performance-unnecessary-value-param';
```

Comment:
If we do in fact need this, I'd prefer hoisting this logic out into the `lint.sh` script to make it more visible.

Reply:
The reason I didn't do that is because it's really ugly doing the same logic in `sh`.  The thought being that having dart code is easier to get right and maintain instead of `sh`.

Reply:
I could make another flag `--mac-host-warnings-as-errors=foobar`.

Reply:
With this PR, these lints are added, like, three layers deep (`.clang-tidy` -> `lint.sh` -> `options.dart`), so I'm more concerned that people are going to get confused about how/why they're enabled than I am about adding logic to the shell script. `--mac-host-warnings-as-errors` is kind of oddly specific, but at least makes the additional easier to find.

Reply:
I agree this is really buried. I think I should merge https://github.com/flutter/engine/pull/36885 and then either

1.  Add these two lints back to the checks when all the platforms issues are resolved, as was done with the other linters.
or 
2. Create override `.clang-tidy` files in the different embedder directories so we can add linter checks to different platforms without needing to fix them everywhere.  I believe `clang-tidy` will prefer the `.clang-tidy` file closest to the file being linted.

Reply:
> --mac-host-warnings-as-errors is kind of oddly specific, but at least makes the additional lints easier to find.

Yea it's weird.  Once we get a solid foothold it should be easier to land the other platforms in another PR while this one holds the line for the most part.  I'll make that change.

> I think I should merge https://github.com/flutter/engine/pull/36885 and then either

I just don't understand how that is helping anyone.  The thesis for removing it was that it was confusing and it wasn't benefiting anyone.  I've shown that it is doing what I expected and it is beneficial in limited cases (albeit not as many as I had hoped).  I agree we need to make it better by turning it into an error on ci.

> Add these two lints back to the checks when all the platforms issues are resolved, as was done with the other linters.

That needlessly makes the task hard for no reason.  In order to do it all at once I need to run clang-tidy-fix 5 times on 3 different platforms, hand fix the automated fix since it isn't 100% perfect, get it all reviewed and landed all within the span of no one introducing another violation.  Otherwise have to start the process all over again.

Having multiple `.clang-tidy` files also has it's own problems where managing multiple files is a pain and the hope is that this is a special case and a transitory one.

Reply:
Done, moved the warnings-as-errors declaration to `lint.sh`.  Leaving the thread unresolved since there was some other comments.

Reply:
> I just don't understand how that is helping anyone. 

It helps because the expected behavior that adding a check to the .clang-tidy file will surface regressions to that lint... somewhere.  Without needing to set up an IDE with clang-tidy, if anyone does this, and without needing to know to run `--fix`.  If we don't merge https://github.com/flutter/engine/pull/36885 then also need to move `modernize-use-default-member-init` to `WarningsAsErrors` as that's a pretty clear example of the confusion. Generally when we add new lints, errors or warnings, in other linter systems, it also involves fixing the existing violations of that lint.  I know these two lints aren't totally cleaned up though on all platforms, and that's the issue here.

However we obviously disagree here, so I hope people notice the behavior of `--mac-host-warnings-as-errors` and any similar platform-specific lints going forward.